process/Style-guide.md

```diff
@@ -0,0 +1,250 @@
+* The canonical pylint configuration for log2timeline project is generated by l2tdevtools. 
+
+
+### Function and method arguments
+
+* Prefix unused function or method arguments with `unused_`. 
+* In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html#Comments) sort keyword arguments alphabetically by name.
+* Call keyword argument with their keyword prefix, not as positional arguments. For example:
+```python
+
+def SomeMethod(self, alternate=False, secondary=True):
+  pass
+
+# No:
+SomeMethod(True, False)
+
+# Yes:
+SomeMethod(secondary=False)
+```
+
+## Strings
+
+* Quote strings as ' or """ and not "
+  * Quote strings in command line arguments (argparse) as "
+* Textual strings should be Unicode strings.
+  * Use the [unicode_literals](http://python-future.org/unicode_literals.html) module to make all strings unicode by default.
+* Use the format() function instead of the %-style of formatting strings.
+  * Use positional or parameter format specifiers with typing e.g. '{0:s}' or '{text:s}' instead of '{0}', '{}' or '{:s}'. If we ever want to have language specific output strings we don't need to change the entire codebase (again). It also makes is easier in determining what type every parameter is expected to be.
+
+## Exceptions
+
+* When catching exceptions use "as exception:" not some alternative form like "as error:" or "as details:"
+* Raise exceptions like this: ```raise MyException('Error message')``` or ```raise MyException```.
+* Although Python allows for ```try ... except ... else``` we prefer not to use it.
+* Make exception messages as useful and descriptive and possible. For example, if an argument is out of an acceptable range, print the invalid value to speed-up debugging.
+
+##Return statements
+
+Per [PEP8](https://pep8.org/#programming-recommendations): "Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)."
+
+* Use `return None` instead of `return` when your function or method is expected to return a value.
+* Do not use `return None` in generators.
+* Use `return` in a function or method that does not return a value.
+
+## Docstrings
+
+* Use English, and avoid abbreviations. Use "for example" or "such as" instead of Latin abbreviations like "e.g.".
+  * Acronyms and initialisms should be preserved, for example HTMLParser and not HtmlParser.
+* We use "Google Style" docstrings. 
+  * For examples, see [this page](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) as well as the notes below. 
+
+There are still a few legacy docstrings in the codebase, here are some examples you might see. Please don't write new code that looks like this:
```

Comment:
Is this still accurate?

Reply:
Possibly, but I don't think it's useful to have all this extra text. I'll clean it up now.